storage/manyrocks/manyrocks.go

```diff
@@ -827,14 +829,67 @@ func (t *Rock) Purge(purgeAddr s.Address) (nextAddr s.Address, nextKey s.Key, er
 			// there's a "change" in the extent content
 			t.notify(s.InvalidKey, s.InvalidAddr)
 
+			type dbgInfo struct {
+				filename string
+				maxAddr  s.Address
+				deleted  bool
+				duration time.Duration
+			}
+
+			var dbg []*dbgInfo
+
+			t0 := time.Now()
+
 			// run through all the SSTs and delete those whose range of
 			// keys are entirely less than the 'purgeAddr'
 
 			for _, file := range t.db.GetLiveFilesMetaData() {
 
+				di := &dbgInfo{filename: file.Name}
+
 				if maxAddr := t.deserializeAddr(file.LargestKey); maxAddr <= purgeAddr {
-					t.store.logger.WithFields(bark.Fields{`Name`: file.Name, `maxAddr`: maxAddr, `purgeAddr`: purgeAddr}).Debug(`PurgeFiles: deleted SST`)
+
+					t0d := time.Now()
 					t.db.DeleteFile(file.Name)
+					di.duration = time.Since(t0d)
+					di.maxAddr = maxAddr
+					di.deleted = true
+
+				} else {
+					di.deleted = false
+				}
+
+				dbg = append(dbg, di)
+			}
+
+			if duration := time.Since(t0); duration > 2*time.Second {
```

Comment:
i think the scale is not very large .. we only log for extents that take a long time and retention runs only once in 10 mins ..